tests/integrations/aws_lambda/client.py

```diff
@@ -68,110 +172,128 @@ def run_lambda_function(
+                    Description="Created as part of testsuite for getsentry/sentry-python",
+                    FunctionName=full_fn_name,
                     Runtime=runtime,
                     Timeout=timeout,
-                    Role=os.environ["SENTRY_PYTHON_TEST_AWS_IAM_ROLE"],
-                    Handler="test_lambda.test_handler",
-                    Code={"ZipFile": zip.read()},
-                    Description="Created as part of testsuite for getsentry/sentry-python",
+                    Role=AWS_LAMBDA_EXECUTION_ROLE_ARN,
+                    Handler=handler,
+                    Code={"ZipFile": lambda_function_zip.read()},
+                    Environment=environment,
+                    Layers=layers,
                 )
-        else:
-            subprocess.run(
-                ["zip", "-q", "-x", "**/__pycache__/*", "-r", "ball.zip", "./"],
-                cwd=tmpdir,
-                check=True,
+
+                waiter = client.get_waiter("function_active_v2")
+                waiter.wait(FunctionName=full_fn_name)
+        except client.exceptions.ResourceConflictException:
+            print(
+                "Lambda function already exists, this is fine, we will just invoke it."
             )
 
-            # Default initial handler
-            if not initial_handler:
-                initial_handler = "test_lambda.test_handler"
+    response = client.invoke(
+        FunctionName=full_fn_name,
+        InvocationType="RequestResponse",
+        LogType="Tail",
+        Payload=payload,
+    )
 
-            build_no_code_serverless_function_and_layer(
-                client, tmpdir, fn_name, runtime, timeout, initial_handler
-            )
+    assert 200 <= response["StatusCode"] < 300, response
+    return response
 
-        @add_finalizer
-        def clean_up():
-            client.delete_function(FunctionName=fn_name)
-
-            # this closes the web socket so we don't get a
-            #   ResourceWarning: unclosed <ssl.SSLSocket ... >
-            # warning on every test
-            # based on https://github.com/boto/botocore/pull/1810
-            # (if that's ever merged, this can just become client.close())
-            session = client._endpoint.http_session
-            managers = [session._manager] + list(session._proxy_managers.values())
-            for manager in managers:
-                manager.clear()
-
-        response = client.invoke(
-            FunctionName=fn_name,
-            InvocationType="RequestResponse",
-            LogType="Tail",
-            Payload=payload,
-        )
 
-        assert 200 <= response["StatusCode"] < 300, response
-        return response
+# This is for inspecting new Python runtime environments in AWS Lambda
+# If you need to debug a new runtime, use this REPL to run arbitrary Python or bash commands
+# in that runtime in a Lambda function:
+#
+#    pip3 install click
+#    python3 tests/integrations/aws_lambda/client.py --runtime=python4.0
+#
```

Comment:
Can't comment on the line directly but on 322 we say "name of the runtime to use, eg python3.8" -- that should probably also be updated to 3.9 if 3.8 is not supported anymore

Reply:
fixed.